test/test_url.rb

```diff
@@ -61,16 +61,6 @@ class TestURL < JekyllUnitTest
       ).to_s
     end
 
-    should "throw an exception if the URL contains a colon" do
-      url = URL.new(
-        :template     => "/:x/:y/:z",
-        :placeholders => { :x => "foo", :z => "bar" }
-      )
-      assert_raises Jekyll::Errors::InvalidURLError do
```

Comment:
What do you and @pathawks think about, if a placeholder doesn't exist, simply removing it?

The real issue we're driving at is: you used an invalid URL placeholder, i.e. `:dat` instead of `:date`.